src/canvas/Arduino_Canvas_Indexed.cpp

```diff
@@ -160,22 +162,27 @@ void Arduino_Canvas_Indexed::raise_mask_level()
         int32_t buffer_size = _width * _height;
         uint8_t old_indexed_size = _indexed_size;
         uint8_t new_color;
+        uint8_t old_to_new_color[sizeof(uint8_t) * 8];   // size 256
         _indexed_size = 0;
         _color_mask = mask_level_list[++_current_mask_level];
         Serial.print("Raised mask level: ");
         Serial.println(_current_mask_level);
 
-        // update _framebuffer color index, it is a time consuming job
+        // update _framebuffer color index, it is (was?) a time consuming job
         for (uint8_t old_color = 0; old_color < old_indexed_size; old_color++)
         {
-            new_color = get_color_index(_color_index[old_color]);
-            for (int32_t i = 0; i < buffer_size; i++)
-            {
-                if (_framebuffer[i] == old_color)
-                {
-                    _framebuffer[i] = new_color;
-                }
-            }
+            uint16_t old_color_indexing = _color_index[old_color];
+            
+            _color_to_index_map.erase(old_color_indexing);      //remove the color with old mask
```

Comment:
This code assumes that there aren't  `old_color` with the same value of `new_color`.

that is, when the _old colors_ are converted to the _new colors_, there will not exist a value of _new colors_ that matches any value of the _old colors_